### PR TITLE
circus: 0.16.1 -> 0.18.0; reintroduce as python-modules

### PR DESCRIPTION
**Pull request by natsukium**
###### Description of changes

Diff: https://github.com/circus-tent/circus/compare/0.16.1...0.18.0

We changed circus to application in #73199, but later updates have removed the bottleneck library version restriction.

I would also like to revert to python-modules, as this package is required for [bentoml](https://github.com/bentoml/BentoML).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kirillrdy** [comment]
Result of `nixpkgs-review pr 243040` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>circus (python310Packages.circus)</li>
    <li>circus.dist (python310Packages.circus.dist)</li>
    <li>python311Packages.circus</li>
    <li>python311Packages.circus.dist</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -6485,7 +6485,7 @@ with pkgs;
 
   circleci-cli = callPackage ../development/tools/misc/circleci-cli { };
 
-  circus = callPackage ../tools/networking/circus { };
+  circus = with python3Packages; toPythonApplication circus;
```
Comment (kirillrdy) [inline pkgs/top-level/all-packages.nix:6488]: no need to change: but maybe someone can clarify why not `python3Packages.toPythonApplication circus`

**Comment by kirillrdy** [comment]
Result of `nixpkgs-review pr 243040` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>circus (python310Packages.circus)</li>
    <li>circus.dist (python310Packages.circus.dist)</li>
    <li>python311Packages.circus</li>
    <li>python311Packages.circus.dist</li>
  </ul>
</details>

@natsukium dont need to fix aarch64-linux issues, just raising for visibility

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (natsukium) [reply pkgs/top-level/all-packages.nix:6488]: Thanks for reviewing this PR.
Without the `with` statement, it should be `python3Packages.toPythonApplication python3Packages.circus`.

**Comment by natsukium** [comment]
Ofborg seems to build it successfully.
Maybe there is a flaky test.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (kirillrdy) [reply pkgs/top-level/all-packages.nix:6488]: ah yes, indeed that makes sense

**Comment by kirillrdy** [comment]
> Ofborg seems to build it successfully. Maybe there is a flaky test

seems to be consistent to me, but if since its good for ofborg, no worries.

Going to take the opportunity to say that I am very impressed with quality your merge requests ( at least the recent ones that I saw )

**Comment by natsukium** [comment]
Thanks!
I'm glad to see that you often fix builds, especially for aarch64-linux.

**Comment by natsukium** [comment]
@happysalada san, if you plan to add bentoml, you will need this.
Would you please review this PR?

**Comment by happysalada** [comment]
This looks good to me, are you planning any other changes ? Just checking, im happy merging this.

**Comment by natsukium** [comment]
It's ready to go.

**Pull request merged**